src/components/ec-input-field/ec-input-field.vue

```diff
@@ -1,14 +1,30 @@
 <template>
-  <div class="ec-input-field">
+  <div
+    class="ec-input-field"
+    data-test="ec-input-field"
+  >
     <label
-      v-if="label"
+      v-if="label || note"
       class="ec-input-field__label"
+      data-test="ec-input-field__label"
       :for="id"
-    >{{ label }}</label>
+    >
+      <span
+        v-if="label"
+        class="ec-input-field__label-text"
+        data-test="ec-input-field__label-text"
+      >{{ label }}</span>
+      <span
+        v-if="note"
+        class="ec-input-field__note"
+        data-test="ec-input-field__note"
+      >{{ note }}</span>
+    </label>
     <input
       :id="id"
       v-model="inputModel"
       class="ec-input-field__input"
+      :data-test="$attrs['data-test'] ? `${$attrs['data-test']} ec-input-field__input` : 'ec-input-field__input'"
```

Comment:
```suggestion
      data-test="ec-input-field__input"
      :data-test="$attrs['data-test']"
```

would achieve the same purpose?

Reply:
This is not working, data-test doesn't work in the same way that the classes and the styles

Good suggestion anyway I didn't now either! :)

Reply:
oops, good to know! not a Vue expert :blush: